mldsa/native/x86_64/src/rej_uniform_avx2.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) The mldsa-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+
+/*
+ * This file is derived from the public domain
+ * AVX2 Dilithium implementation @[REF_AVX2].
+ */
+
+#include "../../../common.h"
+
+#if defined(MLD_ARITH_BACKEND_X86_64_DEFAULT) && \
+    !defined(MLD_CONFIG_MULTILEVEL_NO_SHARED)
+
+#include <immintrin.h>
+#include <stdint.h>
+#include <string.h>
+#include "arith_native_x86_64.h"
+#include "consts.h"
+
+unsigned int mld_rej_uniform_avx2(int32_t *MLD_RESTRICT r, const uint8_t *buf)
```

Comment:
Can you document the desired buffer length, please?

Also, in the upstream AVX2 code I see ominous `+8` in the buffer sizes which suggest some overreading. How is this being dealt with here? Can you confirm that those functions are exercised with asan and minimally allocated buffers, so we would catch overreads?

Reply:
The +8 has been accounted for by adjusting the loop boundary of the first loop. No overreading occurs.
I added a comment explaining the differences and also added the buffer sizes explicitly.